client/src/services/api.service.js

```diff
@@ -0,0 +1,38 @@
+import Axios, { AxiosRequestConfig, AxiosResponse, Method } from "axios";
+
+/**
+ * @type {string}
+ */
+const API = process.env.REACT_APP_API_URL || "http://localhost:8080";
+
+/**
+ * @template T
+ * @param {Method} method
+ * @param {string} url
+ * @param {AxiosRequestConfig} config
+ * @returns {Promise<AxiosResponse<T>>}
+ */
+function request(method, url, config) {
+    return Axios.request({ method, url: `${API}/${url}`, ...config });
+}
+
+class ApiService {
+    /**
+     * @template T
+     * @param {string} url
+     * @param {AxiosRequestConfig} config
+     * @returns {Promise<AxiosResponse<T>>}
+     */
+    get = (url, config) => request("get", url, config);
```

Comment:
The comments here are [JSDoc](https://jsdoc.app/). VSCode is smart enough to recognize and read JSDoc-style comments to produce intellisense. I don't recommend using it everywhere because, as you can see, it's bulky! But it _can_ be helpful for something generic like this that will be used often.

![jsdoc-intellisense](https://user-images.githubusercontent.com/8491377/78242558-2f9b6280-74b0-11ea-8599-493f3468e45d.gif)